src/stan/math/prim/mat/functor/finite_diff_hessian.hpp

```diff
@@ -0,0 +1,110 @@
+#ifndef STAN__MATH__PRIM__MAT__FUNCTOR__FINITE_DIFF_HESSIAN_HPP
+#define STAN__MATH__PRIM__MAT__FUNCTOR__FINITE_DIFF_HESSIAN_HPP
+
+#include <stan/math/prim/mat/fun/Eigen.hpp>
+
+namespace stan {
+  
+  namespace math {
+
+    template <typename F>
+    double
+    finite_diff_hess_helper(const F& f,
+                            const Eigen::Matrix<double,Eigen::Dynamic,1>& x,
+                            const int& lambda,
```

Comment:
Specifically, replace "const int&" with "int".   If
it's a const ref, then it's not being used as a return,
so passing by value is the idiomatic way to go --- it should
be passing something no larger than the reference on 64-bit
architectures.

And then the const doesn't add anything to the external
API, becuase it'll be passed by value.

- Bob

> On Mar 17, 2015, at 5:27 AM, Daniel Lee <notifications@github.com> wrote:
> 
> In src/stan/math/prim/mat/functor/finite_diff_hessian.hpp:
> 
> > @@ -0,0 +1,110 @@
> > +#ifndef STAN__MATH__PRIM__MAT__FUNCTOR__FINITE_DIFF_HESSIAN_HPP
> > +#define STAN__MATH__PRIM__MAT__FUNCTOR__FINITE_DIFF_HESSIAN_HPP
> > +
> > +#include <stan/math/prim/mat/fun/Eigen.hpp>
> > +
> > +namespace stan {
> > +  
> > +  namespace math {
> > +
> > +    template <typename F>
> > +    double
> > +    finite_diff_hess_helper(const F& f,
> > +                            const Eigen::Matrix<double,Eigen::Dynamic,1>& x,
> > +                            const int& lambda,
> 
> remove &
> 
> —
> Reply to this email directly or view it on GitHub.
>